docs/docs/dev_docs/contracts/portals/data_structures.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Data Structures
+---
+
+The `DataStructures` are structs that we are using throughout the message infrastructure and registry.
+
+**Links**: [Implementation](https://github.com/AztecProtocol/aztec-packages/blob/master/l1-contracts/src/core/libraries/DataStructures.sol).
+
+## `Entry`
+
+An entry for the messageboxes multi-sets. 
+
+```solidity title="DataStructures.sol"
+  struct Entry {
+    uint64 fee;
+    uint32 count;
+    uint32 version;
+    uint32 deadline;
+  }
+```
+
+| Name           | Type    | Description |
+| -------------- | ------- | ----------- |
+| `fee`          | `uint64` | The fee provided to sequencer for including in the inbox. 0 if Outbox (as not applicable). |
+| `count`        | `uint32` | The occurrence of the entry in the dataset |
+| `version`      | `uint32` | The version of the entry |
+| `deadline`     | `uint32` | The consumption deadline of the message. |
+
+
+## `L1Actor`
+
+An entity on L1, specifying the address and the chainid for the entity. Used when specifying sender/recipient with an entity that is on L1.
+
+```solidity title="DataStructures.sol"
+  struct L1Actor {
+    address actor;
+    uint256 chainId;
+  }
+```
+
+| Name           | Type    | Description |
+| -------------- | ------- | ----------- |
+| `actor`          | `address` | The ethereum address of the actor |
```

Comment:
```suggestion
| `actor`          | `address` | The L1 address of the actor |
```

Reply:
maybe we should be consistent with using either L1 or ethereum everywhere

Reply:
An `ethereum address` is something specific, not any `l1 address` allowed for format etc, like you can insert a bitcoin address here. But that is being pedantic, so maybe better just with L1 consistently. Will take a look over it and see how often we are using ethereum, think it might just be here.